File: angular-ecommerce/src/app/common/purchase.ts
import {Customer} from "./customer";
import {Address} from "./address";
import {OrderItem} from "./order-item";
import {Order} from "./order";

export class Purchase {
  customer:Customer | null = null;
  shippingAddress:Address | null = null ;
  billingAddress:Address | null = null;
  order:Order | null = null;
  orderItems: OrderItem[] | null = null;



}

File: backend/spring-boot-ecommerce/src/main/java/com/bdelpech/springbootecommerce/dto/Purchase.java
package com.bdelpech.springbootecommerce.dto;

import com.bdelpech.springbootecommerce.entity.Address;
import com.bdelpech.springbootecommerce.entity.Customer;
import com.bdelpech.springbootecommerce.entity.Order;
import com.bdelpech.springbootecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
